## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 453091`
Commit: `28ce7cfc81957692273f48feb0fc17813ebb5e27`

---
### `x86_64-linux`
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.uhashring</li>
    <li>python312Packages.uhashring.dist</li>
    <li>python313Packages.uhashring</li>
    <li>python313Packages.uhashring.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.uhashring</summary>
<pre>    
        def _get_server(self, key):
            if isinstance(key, tuple):
                return self._old_get_server(key)
    
            for i in range(self._SERVER_RETRIES):
                for node in self.uhashring.range(key):
                    if node[&quot;instance&quot;].connect():
                        return node[&quot;instance&quot;], key
    
            return None, None
    
&gt;       memcache = __import__(&quot;memcache&quot;)
                   ^^^^^^^^^^^^^^^^^^^^^^
E       ModuleNotFoundError: No module named &#x27;memcache&#x27;

uhashring/monkey.py:36: ModuleNotFoundError
=========================== short test summary info ============================
FAILED tests/test_monkey.py::test_patch_memcache - ModuleNotFoundError: No module named &#x27;memcache&#x27;
=================== 1 failed, 19 passed, 1 skipped in 0.25s ====================</pre>
</details>
<details>
<summary>python313Packages.uhashring</summary>
<pre>    
        def _get_server(self, key):
            if isinstance(key, tuple):
                return self._old_get_server(key)
    
            for i in range(self._SERVER_RETRIES):
                for node in self.uhashring.range(key):
                    if node[&quot;instance&quot;].connect():
                        return node[&quot;instance&quot;], key
    
            return None, None
    
&gt;       memcache = __import__(&quot;memcache&quot;)
                   ^^^^^^^^^^^^^^^^^^^^^^
E       ModuleNotFoundError: No module named &#x27;memcache&#x27;

uhashring/monkey.py:36: ModuleNotFoundError
=========================== short test summary info ============================
FAILED tests/test_monkey.py::test_patch_memcache - ModuleNotFoundError: No module named &#x27;memcache&#x27;
=================== 1 failed, 19 passed, 1 skipped in 0.23s ====================</pre>
</details>
</details>

--- a/pkgs/development/python-modules/uhashring/default.nix
+++ b/pkgs/development/python-modules/uhashring/default.nix
@@ -1,0 +1,41 @@
+{
+  lib,
+  buildPythonPackage,
+  pytestCheckHook,
+  fetchFromGitHub,
+  hatchling,
+  python-memcached,
+}:
+buildPythonPackage rec {
+  pname = "uhashring";
+  version = "2.4";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "ultrabug";
+    repo = "uhashring";
+    tag = version;
+    hash = "sha256-6zNPExbcwTUne0lT8V6xp2Gf6J1VgG7Q93qizVOAc+k=";
+  };
+
+  build-system = [
+    hatchling
+  ];
+
+  pythonImportsCheck = [
+    "uhashring"
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    python-memcached
+  ];
+
+  meta = {
+    description = "Full featured consistent hashing python library compatible with ketama";
+    homepage = "https://github.com/ultrabug/uhashring";
+    changelog = "https://github.com/ultrabug/uhashring/blob/${src.tag}/CHANGELOG.md";
+    license = lib.licenses.bsd3;
+    maintainers = with lib.maintainers; [ typedrat ];
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -19454,6 +19454,8 @@ self: super: with self; {
 
   uharfbuzz = callPackage ../development/python-modules/uharfbuzz { };
 
+  uhashring = callPackage ../development/python-modules/uhashring { };
+
   uhi = callPackage ../development/python-modules/uhi { };
 
   uiprotect = callPackage ../development/python-modules/uiprotect { };

